runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -4508,6 +4511,25 @@ void J9::X86::TreeEvaluator::transactionalMemoryJITMonitorEntry(TR::Node
       cg->stopUsingRegister(counterReg);
    }
 
+void
+J9::X86::TreeEvaluator::monitorEnterOrExitForValueTypeClass(
+      TR::Node *node,
+      TR::LabelSymbol *snippetLabel,
+      TR::CodeGenerator *cg)
+   {
+   if (cg->isMonitorValueType(node) != TR_maybe)
```

Comment:
Wouldn't it be safer to say == TR_no?

Reply:
This method only handles the case where it's unclear whether the monitor is value type or not. we need to generate extra checks in this case. `== TR_no`is using existing sequence and `== TR_yes` is calling VM helper directly. Both of the later 2 cases wont be handled here.